### ISSUE TITLE
API Module - Lesson

### ISSUE THREAD

**Issue by raysercast1** [code-mentoring/learn#69]
As a user, I want to take a lesson which is going to be inside a path (like lessons in freecodecamp.org). Once I start doing this lesson and complete it. It should be marked as done, so that I can continue my progress to the next lesson.

**Describe the solution you'd like**
An `Lesson` entity in the API that have relation with the User_Path table so that I can keep track of the user's progress inside the path.

**Describe alternatives you've considered**
Possible columns inside the `Lesson` entity:

- Questions
--That are the content or challenges that the user have to complete

- Score(%)
--The amount of challenges the user have done

**Additional context**
e.g.: LIke the lessons in freecodecamp. The completed mark ( ✓ ) and the progress (10/21) showed in this screenshot
![image](https://user-images.githubusercontent.com/54117650/80324182-2ffbf480-87f5-11ea-8d72-944bcb47d0cb.png)

**Blocked by**
- [ ] #83

**Comment by raysercast1** [comment]
@tristanMatthias  @quimgv  IDK if I do it right but this is  what I've been thinking about the lesson entity.
Any suggestion? improve?

**Comment by shaneSheridan** [comment]
@raysercast1 @tristanMatthias 
I'm concerned about implementing as per freecodecamp's lesson design, we might have different content.

I imagine it best to design our own lesson content first, then implement them as per the design.

**Comment by tristanMatthias** [comment]
Agreed, I think we should put this on hold till we have a better idea of the designs.

**Comment by shaneSheridan** [comment]
Created #83 for lesson design.

**Comment by raysercast1** [comment]
@shaneSheridan  @tristanMatthias  hey guys! have a question. When do we can start building this issue? I guess that we have to re-estructure the requerimient, right ?

**Comment by shaneSheridan** [comment]
@raysercast1 @tristanMatthias Looks like development work on "Lesson" related entities are broken down into separate Issues, see #96.

Should this Issue be closed and instead refer to Issues like #96?

**Comment by tristanMatthias** [comment]
Yeah, closing this in favour of other issues
